timelock-corruption-detection/src/main/java/com/palantir/timelock/corruption/detection/HistoryAnalyzer.java

```diff
@@ -16,65 +16,74 @@
 
 package com.palantir.timelock.corruption.detection;
 
-
-import java.util.Comparator;
-import java.util.List;
-import java.util.Optional;
-import java.util.Set;
-import java.util.stream.Collectors;
-
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.Iterables;
 import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.logsafe.Preconditions;
 import com.palantir.paxos.PaxosValue;
 import com.palantir.timelock.history.PaxosAcceptorData;
 import com.palantir.timelock.history.models.CompletePaxosHistoryForNamespaceAndUseCase;
 import com.palantir.timelock.history.models.ConsolidatedLearnerAndAcceptorRecord;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public final class HistoryAnalyzer {
 
-    public static boolean runCorruptionCheckOnHistory(CompletePaxosHistoryForNamespaceAndUseCase history) {
-        return verifyLearnersHaveLearnedSameValues(history)
-        && verifyLearnedValueWasAcceptedByQuorum(history)
-        && verifyLearnedValueIsGreatestAcceptedValue(history);
+    public static List<CorruptionStatus> corruptionStateForNamespaceAndUseCase(
+            CompletePaxosHistoryForNamespaceAndUseCase history) {
+        return Stream.of(learnersHaveLearnedSameValues(history),
+                learnedValueWasAcceptedByQuorum(history),
+                learnedValueIsGreatestAcceptedValue(history))
+                .filter(status -> status != CorruptionStatus.HEALTHY)
+                .collect(Collectors.toList());
```

Comment:
This is empty if everyone is healthy. Is that by design? (If so that's fine, just make sure it is!)

Reply:
Yes, as of now, the plan is to only react to health check violations. So, the list for healthy namespaceAndUse will be empty.